src/main/java/com/internet/shop/web/filter/AuthorizationFilter.java

```diff
@@ -0,0 +1,79 @@
+package com.internet.shop.web.filter;
+
+import static com.internet.shop.controller.LoginController.USER_ID;
+
+import com.internet.shop.library.Injector;
+import com.internet.shop.model.Role;
+import com.internet.shop.model.User;
+import com.internet.shop.service.interfaces.UserService;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class AuthorizationFilter implements Filter {
+    private static final Set<Role.RoleName> USER_ROLE = Set.of(Role.RoleName.USER);
+    private static final Set<Role.RoleName> ADMIN_ROLE = Set.of(Role.RoleName.ADMIN);
+    private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private final UserService userService =
+            (UserService) injector.getInstance(UserService.class);
+    private Map<String, Set<Role.RoleName>> protectedUrls;
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        protectedUrls = new HashMap<>();
+        /*
+        Users only
+        */
+        protectedUrls.put("/product/buy", USER_ROLE);
+        protectedUrls.put("/shopping-cart", USER_ROLE);
+        protectedUrls.put("/shopping-cart/delete", USER_ROLE);
+        protectedUrls.put("/order/complete", USER_ROLE);
```

Comment:
what about /user/orders?

Reply:
I'll change that, but am asking for the explanation, why the admins should not be allowed to have access to the user's order list

Reply:
because your GetUsersOrdersController takes `userId` from the session, it's purpose is just to show orders of the current logged user